op-batcher/batcher/channel_config.go

```diff
@@ -53,25 +53,27 @@ type ChannelConfig struct {
 // value consistent with cc.TargetNumFrames and cc.MaxFrameSize.
 // comprKind can be the empty string, in which case the default compressor will
 // be used.
-func (cc *ChannelConfig) InitCompressorConfig(approxComprRatio float64, comprKind string) {
+func (cc *ChannelConfig) InitCompressorConfig(approxComprRatio float64, comprKind string, compressionAlgo string, compressLevel int) {
 	cc.CompressorConfig = compressor.Config{
 		// Compressor output size needs to account for frame encoding overhead
 		TargetOutputSize: MaxDataSize(cc.TargetNumFrames, cc.MaxFrameSize),
 		ApproxComprRatio: approxComprRatio,
 		Kind:             comprKind,
+		CompressionAlgo:  compressionAlgo,
+		CompressLevel:    compressLevel,
 	}
 }
 
-func (cc *ChannelConfig) InitRatioCompressor(approxComprRatio float64) {
-	cc.InitCompressorConfig(approxComprRatio, compressor.RatioKind)
+func (cc *ChannelConfig) InitRatioCompressor(approxComprRatio float64, compressionAlgo string, compressLevel int) {
+	cc.InitCompressorConfig(approxComprRatio, compressor.RatioKind, compressionAlgo, compressLevel)
 }
 
-func (cc *ChannelConfig) InitShadowCompressor() {
-	cc.InitCompressorConfig(0, compressor.ShadowKind)
+func (cc *ChannelConfig) InitShadowCompressor(compressionAlgo string, compressLevel int) {
+	cc.InitCompressorConfig(0, compressor.ShadowKind, compressionAlgo, compressLevel)
 }
 
-func (cc *ChannelConfig) InitNoneCompressor() {
-	cc.InitCompressorConfig(0, compressor.NoneKind)
+func (cc *ChannelConfig) InitNoneCompressor(compressionAlgo string, compressLevel int) {
+	cc.InitCompressorConfig(0, compressor.NoneKind, compressionAlgo, compressLevel)
```

Comment:
The none compressor doesn't need a level. And if only zlib supports none compression, then also not an algo
```suggestion
func (cc *ChannelConfig) InitNoneCompressor() {
	cc.InitCompressorConfig(0, compressor.NoneKind, derive.Zlib)
```